core/platform/search/elastic_search_services.py

```diff
@@ -212,6 +221,53 @@ def clear_index(index_name: str) -> None:
         })
 
 
+@overload
+def search(
+    query_string: str,
+    index_name: str,
+    categories: List[str],
+    language_codes: List[str],
+    *,
+    offset: Optional[int] = None,
+    size: int = feconf.SEARCH_RESULTS_PAGE_SIZE,
+    ids_only: Literal[True]
+) -> Tuple[List[str], Optional[int]]: ...
+
+
+# Here we use type Any because this method returns the list of document
```

Comment:
@sahiljoster32 Given all this complexity and the fact that we always only use ids_only=True (I think? Please double-check) -- let's just hardcode ids_only=True in the call to elasticsearch and get rid of the other cases.

I don't see us using ids_only=False in the future -- we would always fetch the data from our own datastore anyway.

What do you think? (Also, if ids_only=False is used only in tests and nowhere else, it's fine to convert those tests to use ids_only=True, or drop them if the path they are testing no longer makes sense.)

Reply:
> let's just hardcode ids_only=True in the call to elasticsearch and get rid of the other cases

Done and test cases are also passing!